File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Source/Tva.Phasors/ChannelCellParsingStateBase.vb
'*******************************************************************************************************
'  ChannelCellParsingStateBase.vb - Channel frame cell parsing state base class
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: James R Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  3/7/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

' This class represents the common implementation of the protocol independent parsing state of any kind of data cell.
Public MustInherit Class ChannelCellParsingStateBase

    Inherits ChannelParsingStateBase
    Implements IChannelCellParsingState

    Private m_phasorCount As Int32
    Private m_analogCount As Int32
    Private m_digitalCount As Int32

    Public Property PhasorCount() As Int32 Implements IChannelCellParsingState.PhasorCount
        Get
            Return m_phasorCount
        End Get
        Set(ByVal value As Int32)
            m_phasorCount = Value
        End Set
    End Property

    Public Property AnalogCount() As Int32 Implements IChannelCellParsingState.AnalogCount
        Get
            Return m_analogCount
        End Get
        Set(ByVal value As Int32)
            m_analogCount = Value
        End Set
    End Property

    Public Property DigitalCount() As Int32 Implements IChannelCellParsingState.DigitalCount
        Get
            Return m_digitalCount
        End Get
        Set(ByVal value As Int32)
            m_digitalCount = value
        End Set
    End Property

End Class

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/ChannelCellParsingStateBase.cs
//*******************************************************************************************************
//  ChannelCellParsingStateBase.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  03/07/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents the protocol independent common implementation of the parsing state used by any <see cref="IChannelCell"/>.
    /// </summary>
    public abstract class ChannelCellParsingStateBase : ChannelParsingStateBase, IChannelCellParsingState
    {
        #region [ Members ]

        // Fields
        private int m_phasorCount;
        private int m_analogCount;
        private int m_digitalCount;

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Gets or sets the number of phasor elements associated with the <see cref="IChannelCell"/> being parsed.
        /// </summary>
        public virtual int PhasorCount
        {
            get
            {
                return m_phasorCount;
            }
            set
            {
                m_phasorCount = value;
            }
        }

        /// <summary>
        /// Gets or sets the number of analog elements associated with the <see cref="IChannelCell"/> being parsed.
        /// </summary>
        public virtual int AnalogCount
        {
            get
            {
                return m_analogCount;
            }
            set
            {
                m_analogCount = value;
            }
        }

        /// <summary>
        /// Gets or sets the number of digital elements associated with the <see cref="IChannelCell"/> being parsed.
        /// </summary>
        public virtual int DigitalCount
        {
            get
            {
                return m_digitalCount;
            }
            set
            {
                m_digitalCount = value;
            }
        }

        #endregion
    }
}